src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final int DATE_FORMAT_LENGTH = 10;
+    private static final DateTimeFormatter formatter = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        int[] earnings = new int[names.length];
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            LocalDate currentDate = LocalDate.parse(parts[0], formatter);
+            String currentName = parts[1];
+            int hours = Integer.parseInt(parts[2]);
+            int incomePerHour = Integer.parseInt(parts[3]);
```

Comment:
magic numbers and splitters

Reply:
But if we know that this is exactly how it is and in no other way and is not used anywhere else, then why use constants for this? Especially for parsing parts

Reply:
Because one who reads this code does not know what these numbers mean.